File: support-utils/src/main/java/com/hendraanggrian/support/utils/widget/Toasts.java
package com.hendraanggrian.support.utils.widget;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;
import android.view.View;
import android.widget.Toast;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public final class Toasts {

    private Toasts() {
    }

    @NonNull
    public static Toast showShort(@NonNull Context context, @Nullable CharSequence text) {
        return showText(context, text, Toast.LENGTH_SHORT);
    }

    @NonNull
    public static Toast showShort(@NonNull Context context, @StringRes int resId) {
        return showTextRes(context, resId, Toast.LENGTH_SHORT);
    }

    @NonNull
    public static Toast showShort(@NonNull Context context, @NonNull View view) {
        return showCustomView(context, view, Toast.LENGTH_SHORT);
    }

    @NonNull
    public static Toast showLong(@NonNull Context context, @Nullable CharSequence text) {
        return showText(context, text, Toast.LENGTH_LONG);
    }

    @NonNull
    public static Toast showLong(@NonNull Context context, @StringRes int resId) {
        return showTextRes(context, resId, Toast.LENGTH_LONG);
    }

    @NonNull
    public static Toast showLong(@NonNull Context context, @NonNull View view) {
        return showCustomView(context, view, Toast.LENGTH_LONG);
    }

    @NonNull
    private static Toast showText(@NonNull Context context, @Nullable CharSequence text, int duration) {
        Toast toast = Toast.makeText(context, text, duration);
        toast.show();
        return toast;
    }

    @NonNull
    private static Toast showTextRes(@NonNull Context context, @StringRes int resId, int duration) {
        Toast toast = Toast.makeText(context, resId, duration);
        toast.show();
        return toast;
    }

    @NonNull
    private static Toast showCustomView(@NonNull Context context, @NonNull View view, int duration) {
        Toast toast = new Toast(context);
        toast.setView(view);
        toast.setDuration(duration);
        toast.show();
        return toast;
    }
}

File: kota-core/src/com/hendraanggrian/kota/Toasts.kt
@file:JvmName("Toasts")
@file:Suppress("NOTHING_TO_INLINE", "UNUSED")

package com.hendraanggrian.kota

import android.app.Fragment
import android.content.Context
import android.support.annotation.StringRes
import android.widget.Toast

inline fun Fragment.toast(@StringRes message: Int): Toast = activity.toast(message)
inline fun Context.toast(@StringRes message: Int): Toast = Toast
        .makeText(this, message, Toast.LENGTH_SHORT)
        .apply { show() }

inline fun Fragment.toast(message: CharSequence): Toast = activity.toast(message)
inline fun Context.toast(message: CharSequence): Toast = Toast
        .makeText(this, message, Toast.LENGTH_SHORT)
        .apply { show() }

inline fun Fragment.longToast(@StringRes message: Int): Toast = activity.longToast(message)
inline fun Context.longToast(@StringRes message: Int): Toast = Toast
        .makeText(this, message, Toast.LENGTH_LONG)
        .apply { show() }

inline fun Fragment.longToast(message: CharSequence): Toast = activity.longToast(message)
inline fun Context.longToast(message: CharSequence): Toast = Toast
        .makeText(this, message, Toast.LENGTH_LONG)
        .apply { show() }